azurerm/internal/services/managedapplications/managed_application_resource.go

```diff
@@ -0,0 +1,313 @@
+					"MarketPlace",
+					"ServiceCatalog",
+				}, false),
+			},
+
+			"managed_resource_group_name": azure.SchemaResourceGroupName(),
+
+			"application_definition_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validate.ManagedApplicationDefinitionID,
+			},
+
+			"parameters": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"plan": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+						"product": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+						"publisher": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+						"version": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+						"promotion_code": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+					},
+				},
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceManagedApplicationCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).ManagedApplication.ApplicationClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroupName := d.Get("resource_group_name").(string)
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroupName, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("failed to check for present of existing Managed Application Name %q (Resource Group %q): %+v", name, resourceGroupName, err)
+			}
+		}
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_managed_application", *existing.ID)
+		}
+	}
+
+	parameters := managedapplications.Application{
+		Location: utils.String(azure.NormalizeLocation(d.Get("location"))),
+		Kind:     utils.String(d.Get("kind").(string)),
+		Tags:     tags.Expand(d.Get("tags").(map[string]interface{})),
+	}
+
+	if v, ok := d.GetOk("managed_resource_group_name"); ok {
+		targetResourceGroupId := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s", meta.(*clients.Client).Account.SubscriptionId, v)
+		parameters.ApplicationProperties = &managedapplications.ApplicationProperties{
+			ManagedResourceGroupID: utils.String(targetResourceGroupId),
+		}
+	}
+	if v, ok := d.GetOk("application_definition_id"); ok {
+		a := v.(string)
+		b := utils.String(a)
+		parameters.ApplicationDefinitionID = b
```

Comment:
could we just do this
```suggestion
		parameters.ApplicationDefinitionID = utils.String(v.(string))
```

Reply:
updated